pkg/kubectl/cmd/cmd.go

```diff
@@ -424,6 +424,7 @@ func NewKubectlCommand(in io.Reader, out, err io.Writer) *cobra.Command {
 	addProfilingFlags(flags)
 
 	kubeConfigFlags := genericclioptions.NewConfigFlags()
```

Comment:
Why not just setting that `usePersistentConfig` as part of the constructor?

Reply:
Just as @smarterclayton metioned above, it is a massive change to the contract - we really can't do that. Also it freezes config as well. We should only switch callers that are ok with this kind of persistent config.

Also `genericclioptions.NewConfigFlags()` is a public function, we should not involve any changes that will break downstream applications.

Reply:
genericclioptions is a new thing, we don't have that many consumers, yet. k/k has only 2 uses. It's ok to change this constructor.

Reply:
I think @soltysh isn't proposing changing the behavior silently, but rather adding an option to the constructor so callers have to decide which behavior they want

Reply:
Seems it is safe to change this constructor.

@soltysh @liggitt Updated. PTAL.